jetbrains-core/build.gradle

```diff
@@ -21,6 +33,16 @@ configurations {
     testArtifacts
 }
 
+task generateTelemetry(type: GenerateTelemetry) {
+    inputFiles = []
+    outputDirectory = file("${project.buildDir}/generated-src")
```

Comment:
why not commit the generated files to the tree? 

- makes it more obvious how generated source changes, which helps with code review, git-bisect, human reasoning
- generated output shouldn't have much "churn" so this won't add much noise to VCS history.

Reply:
i don't like the idea of committing generated files - to me VCS is the *source* of files that we own / control / touch / update - it's misleading if you check out a branch and a change in the generator goes and overrides all the of the source-controlled generated files. I only made one change - which is no-where near that - why are all these files changing?!

Reply:
> it's misleading if you check out a branch and a change in the generator goes and overrides all the of the source-controlled generated files. I only made one change

That should never happen, since we pin to a specific version of the generator. 

Certainly build artifacts should not be committed to VCS, and in general I agree. But this generated code is really just scaffolding. So pretending it's not part of the tree is actually risky IMO.